data/mods/desert_region/mapgen/waterbody.json

```diff
@@ -16,194 +16,34 @@
       "river_c_not_se",
       "river_c_not_sw"
     ],
+    "//": "Weight should be removed if/when lake_shore and river (or their weights) are unhardcoded",
     "weight": 100000000000,
     "object": {
       "fill_ter": "t_mudcrack",
       "place_nested": [
-        {
-          "else_chunks": [ "shore_north_west" ],
-          "x": 0,
-          "y": 0,
-          "neighbors": {
-            "north_west": [
-              "lake_bed",
-              "lake_shore",
-              "river_center",
-              "river_nw",
-              "river_ne",
-              "river_sw",
-              "river_se",
-              "river",
-              "river_c_not_ne",
-              "river_c_not_nw",
-              "river_c_not_se",
-              "river_c_not_sw",
-              "bridge",
-              "bridgehead_ground"
-            ]
-          }
-        },
+        { "else_chunks": [ "shore_north_west" ], "x": 0, "y": 0, "flags": { "north_west": [ "BRIDGE", "LAKE", "RIVER" ] } },
```

Comment:
Same thing, this is a good change but should probably be done in another PR, along with a bunch of the related cleanups.

Reply:
That's fair other than that I have no related clean ups I care to do and have 0 interest in doing anything specific with desert_region so it would just be it's own tiny PR. Again if that's desired I'll do it tho.